requirements-dev.txt

```diff
@@ -1,6 +1,6 @@
 -e .[all]
 
-black[jupyter]
```

Comment:
Adding a floor here so that dev environments are consistent.

black has dropped Python 3.7. People shouldn't be doing dev in a 3.7 environment anyways, so skipping black here seems like a good workaround for us to keep the 3.7 tests going for a while.